File: app/src/main/java/net/ducksmanager/persistence/dao/UserDao.java
package net.ducksmanager.persistence.dao;

import net.ducksmanager.persistence.models.dm.User;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(User user);

    @Query("SELECT * FROM users LIMIT 1")
    User getCurrentUser();

    @Query("DELETE FROM users")
    void deleteAll();
}

File: app/src/main/java/net/ducksmanager/persistence/dao/UserDao.kt
package net.ducksmanager.persistence.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import net.ducksmanager.persistence.models.dm.User

@Dao
interface UserDao {
    @get:Query("SELECT * FROM users LIMIT 1")
    val currentUser: User?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(user: User)

    @Query("DELETE FROM users")
    fun deleteAll()
}